monkey/infection_monkey/exploit/powershell_utils/auth_options.py

```diff
@@ -16,15 +16,28 @@ class AuthOptions:
     ssl: bool
 
 
-def get_auth_options(credentials: List[Credentials], use_ssl: bool) -> List[AuthOptions]:
-    auth_options = []
+def get_auth_options(credentials: Credentials, use_ssl: bool) -> AuthOptions:
+    ssl = _get_ssl(credentials, use_ssl)
+    auth_type = _get_auth_type(credentials)
+    encryption = _get_encryption(credentials)
 
-    for creds in credentials:
-        # Passwordless login only works with SSL false, AUTH_BASIC and ENCRYPTION_NEVER
-        ssl = False if creds.secret == "" else use_ssl
-        auth_type = AUTH_BASIC if creds.secret == "" else AUTH_NEGOTIATE
-        encryption = ENCRYPTION_NEVER if creds.secret == "" else ENCRYPTION_AUTO
+    return AuthOptions(auth_type, encryption, ssl)
 
-        auth_options.append(AuthOptions(auth_type, encryption, ssl))
 
-    return auth_options
+def _get_ssl(credentials: Credentials, use_ssl):
+    # Passwordless login only works with SSL false, AUTH_BASIC and ENCRYPTION_NEVER
```

Comment:
Small note: this comment talks about AUTH_BASIC and ENCRYPTION_NEVER, but the code below only concerns ssl.